File: swift/src/Ice/ImplicitContextI.swift
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import IceImpl

class ImplicitContextI: LocalObject<ICEImplicitContext>, ImplicitContext {
    func getContext() -> Context {
        return handle.getContext()
    }

    func setContext(_ newContext: Context) {
        handle.setContext(newContext)
    }

    func containsKey(_ key: String) -> Bool {
        return handle.containsKey(key)
    }

    func get(_ key: String) -> String {
        return handle.get(key)
    }

    func put(key: String, value: String) -> String {
        return handle.put(key, value: value)
    }

    func remove(_ key: String) -> String {
        return handle.remove(key)
    }
}

File: python/python/Ice/ImplicitContextI.py
# Copyright (c) ZeroC, Inc. All rights reserved.

from .ImplicitContext import ImplicitContext

class ImplicitContextI(ImplicitContext):
    def __init__(self, impl):
        self._impl = impl

    def setContext(self, ctx):
        self._impl.setContext(ctx)

    def getContext(self):
        return self._impl.getContext()

    def containsKey(self, key):
        return self._impl.containsKey(key)

    def get(self, key):
        return self._impl.get(key)

    def put(self, key, value):
        return self._impl.put(key, value)

    def remove(self, key):
        return self._impl.remove(key)
